### PR TITLE
docs: union type openapi proposal

### PR DESCRIPTION
**Pull request by lzap**
This is my proposal on how to solve union types in OpenAPI in a friendly manner for kin-openapi library we use.

### CODE REVIEW

File: docs/adr/014-union-type-openapi.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  type: object
+  required:
+    - type
+  properties:
+    type:
+      $ref: '#/components/schemas/UnionTypes'
+    aws:
+      $ref: '#/components/schemas/AWSData'
+    gcp:
+      $ref: '#/components/schemas/AWSData'
```
Comment (ezr-ondrej) [inline docs/adr/014-union-type-openapi.md:70]:
```suggestion
      $ref: '#/components/schemas/GCPData'
```

File: docs/adr/014-union-type-openapi.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    azure:
+      $ref: '#/components/schemas/AWSData'
```
Comment (ezr-ondrej) [inline docs/adr/014-union-type-openapi.md:72]:
```suggestion
      $ref: '#/components/schemas/AzureData'
```

**Review by ezr-ondrej — APPROVED** [review]
Just a nit, :orange_heart: in general :+1:

**Comment by lzap** [comment]
Pushed, thanks.

**Review by ezr-ondrej — APPROVED** [review]

**Pull request merged**